### PR TITLE
[execCommand] fix ReSpec errors

### PR DESCRIPTION
**Pull request by saschanaz**

### CODE REVIEW

File: execCommand.html [hunk 1]
```diff
@@ -639,68 +640,44 @@ <h2 id="methods-to-query-and-execute-commands">Methods to query and
...
-        <p>When an <code><a href=
-        "#editinginputevent">EditingInputEvent</a></code> object is created,
-        the <dfn id="dom-editinginputevent-command" title=
-        "dom-EditingInputEvent-command"><code>command</code></dfn> and <dfn id=
-        "dom-editinginputevent-value" title=
-        "dom-EditingInputEvent-value"><code>value</code></dfn> attributes must
-        both be initialized to the empty string, unless otherwise
-        specified.</p>
+        <p data-dfn-for="EditingBeforeInputEvent">When an <a>
+        EditingBeforeInputEvent</a> object is created, the <dfn>command</dfn>
```
Comment (marcoscaceres) [inline execCommand.html:673]: can probably use `{{EditingBeforeInputEvent}}` here, but `<a>` is also fine.

File: execCommand.html [hunk 2]
```diff
@@ -639,68 +640,44 @@ <h2 id="methods-to-query-and-execute-commands">Methods to query and
...
+        and <dfn>value</dfn> attributes must both be initialized to the empty
+        string, unless otherwise specified.</p>
+
+        <p data-dfn-for="EditingInputEvent">When an <a>EditingInputEvent
+        </a> object is created, the <dfn>command</dfn> and <dfn>value</dfn>
+        attributes must both be initialized to the empty string, unless
```
Comment (marcoscaceres) [inline execCommand.html]: Unrelated, but I wonder if the IDL should just EventInputEventInitDict should default the corresponding members to `= ""`. Initializing them in prose is not ideal.

File: execCommand.html [hunk 3]
```diff
@@ -795,8 +772,7 @@ <h2 id="methods-to-query-and-execute-commands">Methods to query and
                         <a class="external" data-anolis-spec="dom" href=
                         "http://dom.spec.whatwg.org/#concept-event" title=
                         "concept-event">event</a> at <var title="">affected
```
Comment (marcoscaceres) [inline execCommand.html]:
```suggestion
                        "concept-event">event</a> at <var>affected
```

File: execCommand.html [hunk 4]
```diff
@@ -795,8 +772,7 @@ <h2 id="methods-to-query-and-execute-commands">Methods to query and
...
-                        editing host</var> that uses the <code><a href=
-                        "#editingbeforeinputevent">EditingBeforeInputEvent</a></code>
+                        editing host</var> that uses the <a>EditingBeforeInputEvent</a>
                         interface. The event's <code class="external"
                         data-anolis-spec="dom" title="dom-Event-type"><a href=
```
Comment (marcoscaceres) [inline execCommand.html]: heh, unrelated, but `data-anolis-spec`:) wow, this spec is ooooold school!

Comment (saschanaz) [reply execCommand.html]: OMG what did it even mean??

File: execCommand.html [hunk 5]
```diff
@@ -639,68 +640,44 @@ <h2 id="methods-to-query-and-execute-commands">Methods to query and
...
+
+            [Exposed=Window,
+             Constructor(DOMString type, optional EditingInputEventInit eventInitDict = {})]
+            interface EditingInputEvent : Event {
+              readonly attribute DOMString command;
+              readonly attribute DOMString value;
+            };
+            
+            dictionary EditingInputEventInit : EventInit {
+              DOMString command;
```
Comment (marcoscaceres) [inline execCommand.html:663]:
```suggestion
              DOMString command = "";
```

File: execCommand.html [hunk 6]
```diff
@@ -639,68 +640,44 @@ <h2 id="methods-to-query-and-execute-commands">Methods to query and
...
+              DOMString value;
```
Comment (marcoscaceres) [inline execCommand.html:664]:
```suggestion
              DOMString value = "";
```

File: execCommand.html (hunk 4 above)
Comment (marcoscaceres) [reply execCommand.html]: Anolis was the thing on which ReSpec was originally based... it's what we used in the early 2000s before ReSpec and  BikeShed :)

File: execCommand.html (hunk 2 above)
Comment (saschanaz) [reply execCommand.html]: Can the prose be removed then, for brevity?

Comment (marcoscaceres) [reply execCommand.html]: Absolutely!

File: execCommand.html (hunk 3 above)
Comment (saschanaz) [reply execCommand.html]: Could you restore the preceding space?

File: execCommand.html (hunk 5 above)
Comment (saschanaz) [reply execCommand.html:663]: Committing the suggestion somehow didn't work, so I did them manually.

File: execCommand.html (hunk 3 above)
Comment (saschanaz) [reply execCommand.html]: `var title=""` is everywhere as `data-anolis-spec` is. Maybe in another cleanup?

Comment (marcoscaceres) [reply execCommand.html]: yeah, we can get rid of all of the `data-anolis-spec` stuff... separate PR?

**Comment by marcoscaceres** [comment]
@saschanaz, could you also file a bug on the Gecko side to update the IDL? You should be able to find the IDL file in `dom/webidl`. The Component should be "Core :: DOM: Core & HTML". 

If you'd like to have a go at fixing that on the Gecko side, I can mentor you (or review it if actual code changes are needed). It should be pretty easy to update.

File: execCommand.html (hunk 3 above)
Comment (saschanaz) [reply execCommand.html]: It seems the spec needs some serious cleanup as there are so many obsolete things, like mentioning Firefox 6.0a1 😄

Comment (marcoscaceres) [reply execCommand.html]: lol, yeah :)... it needs a little love. Feel free to send patches.

**Comment by saschanaz** [comment]
> @saschanaz, could you also file a bug on the Gecko side to update the IDL? You should be able to find the IDL file in `dom/webidl`. The Component should be "Core :: DOM: Core & HTML".
> 
> If you'd like to have a go at fixing that on the Gecko side, I can mentor you (or review it no actual code changes are needed). It should be pretty easy to update.

I think we don't need it, because we have no actual implementation.

**Comment by marcoscaceres** [comment]
Lol oops 😬 ok.

**Comment by saschanaz** [comment]
If we are to maintain the spec, and given that the feature itself is going to be replaced by `contentEditable` and `InputEvent`, maybe better to remove the long-unimplemented part and just keep the implemented ones.

**Comment by marcoscaceres** [comment]
> better to remove the long-unimplemented part and just keep the implemented ones.

@johanneswilm wdyt? Or are the those things prollyfilled by libraries? Might be nice to have a list of things that could potentially be removed.

**Comment by johanneswilm** [comment]
The document was originally written when it was it was believed that `execCommand` was the way forward and that it would expand over time and the document was to recommend the way forward. That explains how there are 0 implementation things in there.

Those days are long gone. Most of `execCommand` has been dead weight in browsers for a long time, while a few parts (mostly related to clipboards and until recently to turn off table controls and object resizing on Firefox) are still in use.

So yes, I'm very much in favor of cutting anything that is not implemented anywhere. The document should aim at only helping implementers to standardize those parts that  are being shipped and not add anything extra.

**Comment by marcoscaceres** [comment]
Excellent! thanks for confirming @johanneswilm. 

@saschanaz, if you spot any candidates to chop and want to send some PRs, that would be awesome.

**Review by marcoscaceres — APPROVED** [review]
Approving based on just fixing the ReSpec errors.

**Comment by saschanaz** [comment]
I think EditingBeforeInputEvent and EditingInputEvent covered here are in the list of those candidates 😀

**Comment by marcoscaceres** [comment]
Ok, let's merge this for fun... then new PRs to take them out.

**Pull request merged**